unit_tests/src/test_connection_pool.cpp

```diff
@@ -107,5 +113,129 @@ TEST_CASE("connection pool")
 
         REQUIRE(released_conn_ptr);
     }
-}
 
+    SECTION("connection pools allow the default authentication method to be overridden")
+    {
+        REQUIRE_NOTHROW([&env] {
+            const auto auth_func = [](auto& _comm)
+            {
+                if (clientLogin(&_comm) != 0) {
+                    THROW(SYS_SOCK_CONNECT_ERR, ".irodsA authentication error");
+                }
```

Comment:
This is identical to what the connection pool does, but it demonstrates that it can be changed.